### PR TITLE
[xcbuild] Check local user path for xcsdk configuration

### PR DESCRIPTION
**Pull request by Ktwu**
Depending on whether the tool is run under root or the local user, check paths appropriate for each.

### CODE REVIEW

File: Libraries/xcsdk/Headers/xcsdk/Configuration.h [hunk 1]
```diff
@@ -46,14 +46,14 @@ class Configuration {
...
+     * The default paths for the configuration.
      */
-    static std::string DefaultPath();
+    static std::vector<std::string> DefaultPaths();
 
     /*
      * Loads a configuration from the filesystem.
      */
-    static ext::optional<Configuration> Load(libutil::Filesystem const *filesystem, std::string const &path);
+    static ext::optional<Configuration> Load(libutil::Filesystem const *filesystem, std::vector<std::string> const &path);
```
Comment (grp) [inline Libraries/xcsdk/Headers/xcsdk/Configuration.h]: nit: paths

File: Libraries/xcsdk/Sources/Configuration.cpp [hunk 2]
```diff
@@ -53,10 +61,16 @@ LoadArray(plist::Array const *array)
 }
 
 ext::optional<Configuration> Configuration::
-Load(Filesystem const *filesystem, std::string const &path)
+Load(Filesystem const *filesystem, std::vector<std::string> const &paths)
 {
     std::vector<uint8_t> contents;
-    if (!filesystem->read(&contents, path)) {
+    for (std::string path : paths) {
```
Comment (grp) [inline Libraries/xcsdk/Sources/Configuration.cpp]: nit: `std::string const &path : paths`, avoid a copy

**Comment by grp** [comment]
Looks like the tests need to be updated too?

File: Libraries/xcsdk/Headers/xcsdk/Configuration.h (hunk 1 above)
Comment (Ktwu) [inline Libraries/xcsdk/Headers/xcsdk/Configuration.h]: done

File: Libraries/xcsdk/Sources/Configuration.cpp (hunk 2 above)
Comment (Ktwu) [inline Libraries/xcsdk/Sources/Configuration.cpp]: done

File: Libraries/xcsdk/Sources/Configuration.cpp [hunk 3]
```diff
@@ -26,14 +28,20 @@ Configuration(
...
 {
+    std::vector<std::string> defaultPaths;
     if (char const *environmentPath = getenv("XCSDK_CONFIGURATION_PATH")) {
-        return std::string(environmentPath);
+        defaultPaths.push_back(std::string(environmentPath));
     } else {
-        return "/var/db/xcsdk_configuration.plist";
+        char const *homePath = getenv("HOME");
+        if (getuid() != 0 && homePath != nullptr) {
+            defaultPaths.push_back(std::string(homePath) + "/.xcsdk/xcsdk_configuration.plist");
```
Comment (grp) [inline Libraries/xcsdk/Sources/Configuration.cpp]: Why not just `~/.xcsdk_configuration.plist`?

Comment (Ktwu) [inline Libraries/xcsdk/Sources/Configuration.cpp]: No particular strong reason.  My team does something similar with organizing custom Buck files -- we store data inside of ~/.buckd/...

**Pull request closed without merging**